boards/arm/nucleo_l552ze_q/nucleo_l552ze_q_defconfig

```diff
@@ -0,0 +1,45 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_SOC_SERIES_STM32L5X=y
+CONFIG_SOC_STM32L552XX=y
+# 110MHz system clock
+CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=110000000
+
+# enable uart driver
+CONFIG_SERIAL=y
+
+# enable pinmux
+CONFIG_PINMUX=y
+
+# enable GPIO
+CONFIG_GPIO=y
+
+# clock configuration
+CONFIG_CLOCK_CONTROL=y
+# SYSCLK selection
+CONFIG_CLOCK_STM32_SYSCLK_SRC_PLL=y
+# PLL configuration
+CONFIG_CLOCK_STM32_PLL_SRC_MSI=y
+CONFIG_CLOCK_STM32_MSI_RANGE=6
+#CONFIG_CLOCK_STM32_LSE=y
+#CONFIG_CLOCK_STM32_MSI_PLL_MODE=y
+# produce 110MHz clock at PLL output
+CONFIG_CLOCK_STM32_PLL_M_DIVISOR=1
+CONFIG_CLOCK_STM32_PLL_N_MULTIPLIER=55
+CONFIG_CLOCK_STM32_PLL_P_DIVISOR=7
+CONFIG_CLOCK_STM32_PLL_Q_DIVISOR=2
+CONFIG_CLOCK_STM32_PLL_R_DIVISOR=2
+CONFIG_CLOCK_STM32_AHB_PRESCALER=1
+CONFIG_CLOCK_STM32_APB1_PRESCALER=1
+CONFIG_CLOCK_STM32_APB2_PRESCALER=1
+
+
+# console
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+
+# Enable MPU
+CONFIG_ARM_MPU=y
+
+# This Board implies building Non-Secure firmware
+CONFIG_TRUSTED_EXECUTION_NONSECURE=y
```

Comment:
@erwango I wonder about this: if you set _NONSECURE by default, one cannot use this platform as standalone in Zephyr, e.g. to run kernel tests, CI samples, etc.: that's because a non-secure firmware requires a secure boot.
Similar V8-M platforms in Zephyr by other vendors have introduced both secure and non-secure target definitions.

I wonder if the STM32 should also align with this.

Reply:
@ioannisg by default, TZ is disabled so, we can run zephyr samples on the platform.
I push secure target definitions (basically '_s' version of dts), and we'll be able to run zephyr in secure mode with TZEN=1.
Though, as I understood, in order to be used with TFM (platform upstream is on going on TFM project), tosay zephyr should be run on non secure side. So I didn't planned to upstream secure target definition in zephyr.

Reply:
You are defining here `CONFIG_TRUSTED_EXECUTION_NONSECURE=y`.
This implies that the Zephyr image is meant to execute in the Non-Secure domain.
This Kconfig is used in a few places in CMake and inside the code where we need to have awareness that the execution domain is non-secure.
Therefore, trying to flash this Nucleo board with the generated image as a _standalone, secure-by-default_ image may lead to problems, e.g. undefined behavior. 

My suggestion is to follow the practice of other TrustZone-enabled platforms in the tree, that is, define two Board definitions, one default, which will be used to run, for example, samples and tests, and one _ns_ target that is meant to run in non-secure mode, side-by-side with some secure firmware, e.g. TF-M. This is done for ARM, Nordic and NXP vendor platforms.

If you don't wish to follow this practice, and want the introduced board to be a strictly non-secure image, that is also fine, but I suggest you state that clearly in the documentation of the board. E.g. saying that flashing the image as-is into an empty Nucleo board does not guarantee to work properly with the zephyr samples' and tests' suite.

Reply:
> My suggestion is to follow the practice of other TrustZone-enabled platforms in the tree, that is, define two Board definitions, one default, which will be used to run, for example, samples and tests, and one ns target that is meant to run in non-secure mode, side-by-side with some secure firmware, e.g. TF-M. This is done for ARM, Nordic and NXP vendor platforms.

IIUC, proposed strategy:
- defines a secure target config, that will be used for tests but not used in applications
- defines a non-secure target, that will be used in applications, but not tested

> E.g. saying that flashing the image as-is into an empty Nucleo board does not guarantee to work properly with the zephyr samples' and tests' suite.

Why wouldn't it be possible to consider TZEN=0 as a valid configuration that allows to run zephyr samples & tests?

Reply:
> defines a secure target config, that will be used for tests but not used in applications

well, it could be used in applications that do not require trust-zone; but, yes, the main point is to have regression testing

>Why wouldn't it be possible to consider TZEN=0 as a valid configuration that allows to run zephyr samples & tests?

I don't understand what TZEN=0 means; is TZEN some macro in the code?

My point is that with NONSECURE_FIRMWARE=y would be bringing build-time configuration different from default. For example, certain parts of CMSE module are not compiled-in, same in fault.c - that means that core ARCH code may behave differently and you might notice differences in tests/samples compared to other v8-m targets.

Btw, have you tried to run the full kernel and arm test suites on this target successfully?

Reply:
> I don't understand what TZEN=0 means; is TZEN some macro in the code?

TZEN is an option bit that can be configured by end user. 
Option bits are available to the user to configure hardware options in the controller (like single or dual-bank flash layout or watchdog behavior, but also flash memory protection to configure the mcu as a final product). This is generally done using a tool (STM32CubeProgrammer).

Back to TZEN: it is used to control TrustZone element.
"When the TrustZone is deactivated (TZEN is modified from 1 to 0) after option bytes
loading, the following security features are deactivated:
• Watermark-based secure area
• Block-based secure area
• RDP level 0.5
• Secure interrupt
• All secure registers are RAZ/WI."
By default, on nucleo_l552ze_q, TZEN is set to 0. It is up to the user to modify this and modify TZEN value to enable TrustZone.

You'll find more info in https://www.st.com/resource/en/reference_manual/dm00346336-stm32l552xx-and-stm32l562xx-advanced-armbased-32bit-mcus-stmicroelectronics.pdf

So basically, what I'm pushing is this default out of the box config, with TrustZone disabled.

Regarding kernel tests, 4 are failling:
- tests/kernel/timer/timer_api/kernel.timer.tickless
- tests/kernel/timer/timer_api/kernel.timer
- tests/kernel/mem_protect/userspace/kernel.memory_protection.userspace.gap_filling
- tests/kernel/mem_protect/mem_protect/kernel.memory_protection.gap_filling

I had no time to analyze yet, but I know we have some stability issue on the timer tests, so this may not be M33 specific. I'll have a closer look to the 2 last ones.

Reply:
I don't think the failures in these tests have much to do with TrustZone

Reply:
@ioannisg aside, from the failing tests, do you have concerns in this PR?
Given the information I provided, what would you advise regarding NONSECURE_FIRMWARE?